Add extensions

JinGyuriSin opened the pull request:
새로운 익스텐션 추가

Commit:
[add]
- Main ViewModel 추가
- 나가기 api 구현
- id 저장 기능 구현
- Main 애니메이션 구현중

[change]
- 통신모듈 Status Code 처리하도록 구현

Commit:
[change]
- 통신모듈에서 통신시 Progress 표시하도록 수정

Commit:
[fix]
- message 처리 메인스레드에서 하도록 수정

Commit:
[change]
- 통신시 progressBar 및 메세지는 자동으로 처리되지 않게 변경

Commit:
[add]
- Toast, Snackbar, Intent, Dialog 익스텐션 추가

[change]
- startActivity 호출방식 변경

[remove]
- BaseActivity 내의 함수 삭제

Xellsky commented:
Toast 띄우는 함수 이름을 Snackbar 와 마찬가지로 showToast(...) 식으로 쓰면 어떨까요? 함수명 떠올리기도 쉽고 일관성도 유지되니까요. short, long 구분은 옵션 파라미터로 넣어서 선택할 수 있도록 하면 좋을 것 같네요.

Commit:
[fix]
- 코틀린 버전 업데이트시 Type inference error 해결을 위해 toMap 의 리턴타입 조정

[edit]
- Toast 익스텐션 파라메터로 Duration 받는걸로 수정
- Toast 네이밍 showToast 로 수정
- Intent 익스텐션 내부 이름 수정

JinGyuriSin commented:
@Xellsky 알렉스의 의견을 반영하였습니당

Ethan3972 commented:
animPay 같은 경우 단순한 좌표 이동 애니메이션을 Pay에 제한하여 사용될 것으로 보이는데 
duration도 좌표와 함께 넣어서 animMove로 만들어 여러 곳에 재활용이 가능하게 하는 건 어떤가요..?
